File: C++/647. Palindromic Substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.size();
        int num = 0;
        for(int i = 0; i < n; i++) {
            palindromic(s, i, i, num);  
            palindromic(s, i, i+1, num);
        }
        return num;
    }    
private:
    void palindromic(string s, int left, int right, int& num) { 
        while(left >=0 && right < s.size() && s[left] == s[right]) {
            num++;
            left--;
            right++;
        }
    }
};

File: JAVA/647. Palindromic Substrings.java
class Solution {
public int countSubstrings(String s) {
     int res = 0;
for (int i = 0; i < s.length(); ++i) {
  res += extendPalindromes(s, i, i);
  res += extendPalindromes(s, i, i + 1);
            }
            return res;
           }
private int extendPalindromes(final String s, int l, int r) {
          int count = 0;
     while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            ++count;
            --l;
             ++r;
          }
 return count;
        }
       }
